call.go

```diff
@@ -95,6 +95,9 @@ func sendRequest(ctx context.Context, dopts dialOptions, compressor Compressor,
 			Client: true,
 		}
 	}
+	if compressor == nil && c.compressorType != "" && encoding.GetCompressor(c.compressorType) == nil {
```

Comment:
Not Done. I only removed part from client. Server will receive a string of compressor name and it may not be set by UseCompressor.
Also, for the end2end_test.go/TestCompressServerHasNoSupport, if the server register "gzip" during init(), this test won't pass. Do I need to copy gzip compressor and name it "gzip_tmp" for this test?